diff --git a/setup.py b/setup.py
index 8251449..5df441f 100644
--- a/setup.py
+++ b/setup.py
@@ -327,6 +327,9 @@ def setup_args():
     add_lib("tango", directories, sys_libs, lib_name="tango")
     add_lib("omni", directories, sys_libs, lib_name="omniORB4")
     add_lib("zmq", directories, sys_libs, lib_name="libzmq")
+    add_lib("cppzmq", directories, sys_libs)
+    if "cppzmq" in directories["libraries"]:
+        directories["libraries"].remove("cppzmq")
     add_lib_boost(directories)
 
     macros = []
